src/mscorlib/shared/System/IO/PathHelper.Windows.cs

```diff
@@ -13,144 +12,34 @@ namespace System.IO
-                                break;
-                            case '\\':
-                                segmentLength = index - lastSeparator - 1;
-                                lastSeparator = index;
-
-                                if (foundTilde)
-                                {
-                                    if (segmentLength <= MaxShortName)
-                                    {
-                                        // Possibly a short path.
-                                        possibleShortPath = true;
-                                    }
-
-                                    foundTilde = false;
-                                }
-
-                                if (possibleBadUnc)
-                                {
-                                    // If we're at the end of the path and this is the first separator, we're missing the share.
-                                    // Otherwise we're good, so ignore UNC tracking from here.
-                                    if (index == fullPath.Length - 1)
-                                        throw new ArgumentException(SR.Format(SR.Arg_PathIllegalUNC_Path, fullPath.ToString()));
-                                    else
-                                        possibleBadUnc = false;
-                                }
-
-                                break;
-
-                            default:
-                                if (checkInvalidCharacters && current < ' ') throw new ArgumentException(SR.Argument_InvalidPathChars, nameof(path));
-                                break;
-                        }
-                    }
-
-                    index++;
-                }
-
-                if (possibleBadUnc)
-                    throw new ArgumentException(SR.Format(SR.Arg_PathIllegalUNC_Path, fullPath.ToString()));
-
-                segmentLength = fullPath.Length - lastSeparator - 1;
-
-                if (foundTilde && segmentLength <= MaxShortName)
-                    possibleShortPath = true;
-
-                // Check for a short filename path and try and expand it. Technically you don't need to have a tilde for a short name, but
-                // this is how we've always done this. This expansion is costly so we'll continue to let other short paths slide.
-                if (expandShortPaths && possibleShortPath)
+                if (fullPath.Contains('~'))
                 {
                     return TryExpandShortFileName(ref fullPath, originalPath: path);
                 }
                 else
                 {
                     if (fullPath.Length == path.Length && fullPath.StartsWith(path))
                     {
-                        // If we have the exact same string we were passed in, don't bother to allocate another string from the StringBuilder.
+                        // If we have the exact same string we were passed in, don't bother to allocate another string from the StringBuffer.
```

Comment:
Nit, no need to fix, but if code always returns out of a conditional block
```c#
if (...)
{
    // ..
    return ...;
}
else
{
  . ...
}
```

I think it's easier to read if you avoid the indent/braces ie
```c#
if (...)
{
    // ..
    return ...;
}
  . ...
```

Reply:
I agree. I usually early out.